zarr/indexing.py

```diff
@@ -52,6 +52,8 @@ def is_scalar(value, dtype):
         return True
     if isinstance(value, tuple) and dtype.names and len(value) == len(dtype.names):
         return True
+    if dtype.kind == "O" and not isinstance(value, np.ndarray):
```

Comment:
This is the core issue in #1874. We need to treat `[1, 2]` as a scalar for object arrays.

Is this the right condition?

cc @jakirkham @aplowman